worker/uniter/pebblepoller_test.go

```diff
@@ -141,3 +144,42 @@ func (c *fakePebbleClient) CloseIdleConnections() {
 	defer c.mut.Unlock()
 	c.closed = true
 }
+
+// AddNotice adds a notice for WaitNotices to receive. To have WaitNotices
+// return an error, use notice.Type "error" with the message in notice.Key.
+func (c *fakePebbleClient) AddNotice(checkC *gc.C, notice *pebbleclient.Notice) {
+	select {
+	case c.noticeAdded <- notice:
+	case <-time.After(time.Second):
+		checkC.Fatalf("timed out waiting to add notice")
+	}
+}
+
+func (c *fakePebbleClient) WaitNotices(ctx context.Context, serverTimeout time.Duration, opts *pebbleclient.NoticesOptions) ([]*pebbleclient.Notice, error) {
+	timeoutCh := c.clock.After(serverTimeout)
+	for {
+		select {
+		case notice := <-c.noticeAdded:
+			if notice.Type == "error" {
+				return nil, errors.New(notice.Key)
+			}
+			if noticeMatches(notice, opts) {
+				return []*pebbleclient.Notice{notice}, nil
+			}
+		case <-timeoutCh:
+			return nil, nil // no notices after serverTimeout is not an error
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	}
+}
+
+func noticeMatches(notice *pebbleclient.Notice, opts *pebbleclient.NoticesOptions) bool {
+	if opts == nil || opts.Types != nil || opts.Keys != nil {
+		panic("not supported")
```

Comment:
Can't we pass in a C and assert on that instead of panicing?

Reply:
Not easily. This is called by the implementation of `WaitNotices` which is fulfilling an interface and so we can't adjust the parameters being passed. We *could* pass the `C` into the `fakePebbleClient` struct, but that feels weird. The test fails with a panic in the case this happens, which it shouldn't -- it's really just an assertion.